extensions/ql-vscode/test/vscode-tests/utils/mocking.helpers.ts

```diff
@@ -0,0 +1,16 @@
+export type DeepPartial<T> = T extends object
+  ? {
+      [P in keyof T]?: DeepPartial<T[P]>;
+    }
+  : T;
+
+export function mockedObject<T extends object>(methods: DeepPartial<T>): T {
+  return new Proxy<T>({} as unknown as T, {
+    get: (_target, prop) => {
+      if (prop in methods) {
+        return (methods as any)[prop];
```

Comment:
I was hoping that the TS compiler would work here, but it's giving this error: `TS7053: Element implicitly has an 'any' type because expression of type 'string | symbol' can't be used to index type '{}'.   No index signature with a parameter of type 'string' was found on type '{}'.` I'm not sure how to fix that, so casting `as any` seems the easiest option here.